[KaKa0 adult] 장바구니 업데이트 기능 구현

hyo-jin-jeong opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
- 장바구니 업데이트 기능 구현
		 
## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)
- 장바구니 count update 기능
- 장바구니 select 값 변경 기능

## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)
- 경훈님! 장바구니 count update 했던 부분에 select 값 변경도 patch 함수 하나로 해결하는 것이 좋을거 같아서 두개 한번에 구현했는데, 괜찮을까요??

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

hyo-jin-jeong committed:
Add: 장바구니 수량 업데이트 메소드 구현
- patch method 구현

hyo-jin-jeong committed:
Modify: urls.py 변경

hyo-jin-jeong committed:
Modify: path 메소드 return 추가

hyo-jin-jeong committed:
Modify: count if문 조건 변경

hyo-jin-jeong committed:
Refactor: 불필요한 코드 제거

hyo-jin-jeong committed:
Modify: count if문 조건 변경

hyo-jin-jeong committed:
Modify: 공백 추가

hyo-jin-jeong committed:
Modify: 오타 수정

hyo-jin-jeong committed:
Merge: main브랜치와 병합 후 충돌 해결

hyo-jin-jeong committed:
Merge: main브랜치와 병합 후 충돌 해결

hyo-jin-jeong committed:
Refactor: 공백 제거

hyo-jin-jeong committed:
Refactor: 공백 제거

hyo-jin-jeong committed:
Add: 장바구니 체크 flag 변경 기능 추가

hyo-jin-jeong committed:
Modify: 오류 수정 bool -> strtobool 추가

Kang-kyunghun commented on orders/views.py:
```diff
@@ -74,4 +75,31 @@ def get(self, request):
+        try:
+            data          = json.loads(request.body)
+            order_item_id = data['order_item_id']
+            count         = data.get('count')
+            select        = data.get('select')
```
select는 front한테 어떤 값으로 받고 있나요? 0 또는 1 아니면 True/ False로 무조건 받으면 조금 더 좋을 것 같아요.
(아마 True/False는 JS랑 조금 달라서 힘들 것 같고 0/1 로 받으면 되지 않을까 싶네요)

Kang-kyunghun replied:
```suggestion
            count         = data['count']
            select        = data['select']
```
무조건 필수로 받는 것은 어떨까요? 정보를 수정하는 거니까 수정해야 하는 정보들은 필수로 받으면 좋을 것 같아요.
필수로 받게 되면서 밑에 if 들이 어떻게 바뀔 수 있을지 생각해 보고 수정해 보세요.

hyo-jin-jeong committed:
Refactor: select 값 문자열 true/false -> 1/0 변경

Kang-kyunghun commented on orders/views.py:
```diff
@@ -74,4 +75,31 @@ def get(self, request):
+            if not OrderItem.objects.filter(order__user=request.user, id=order_item_id).exists():
+                return JsonResponse({'message':'INVALID_ORDER_ITEM'}, status=400)
+            
+            order_item = OrderItem.objects.filter(id=order_item_id)
```
```suggestion
            order_item = OrderItem.objects.get(id=order_item_id)
```
get을 이용해서 데이터를 특정 짓는 것이 더 좋습니다~

hyo-jin-jeong committed:
Update orders/views.py

그 프론트쪽에서 count 값을 올릴때랑 select를 바꿀때랑 body를 다르게 주고 있는데 둘다 같은 body값을 사용하는게 괜찮을까요?.. 요청 버튼이 다릅니다! 지금은 count 올릴땐 count 값이 오고 select를 바꿀댄 select만 보내줍니다

Co-authored-by: Kang-kyunghun <68374847+Kang-kyunghun@users.noreply.github.com>

hyo-jin-jeong committed:
Modify: filter() -> get() 변경

hyo-jin-jeong committed:
merge main

hyo-jin-jeong committed:
Merge branch 'feature/basket-update' of https://github.com/wecode-bootcamp-korea/21-1st-KaKa0Adult-backend into feature/basket-update

Kang-kyunghun commented:
LGTM~!!!
